File: TaskOOP/person/designer.py
from person.human import Human

class Designer(Human):

    def __init__(self, firstName, sName, salary, exp, coef, manager):
        Human.__init__(self, firstName, sName, salary, exp, manager)
        self.coef = coef

    def getEmployee(self):
        print ("{0} {1}, manager: {2}, experience: {3}, coeficient {4}".format(self.firstName, self.sName, self.manager.sName,
                                                        self.exp, self.coef))

    def sal(self):
        return self.salary*self.coef

    def printSallary(self):
        print("Designer", self.firstName, self.sName, "has salary", self.sal())

File: OOPJava/src/main/java/exercise1/person/Designer.java
package exercise1.person;

public class Designer extends Employee {
    protected double coefficient;
    public Designer(String firstName, String secondName, int salary, int experience, Manager manager, double coefficient) {
        super(firstName, secondName, salary, experience, manager);
        if (coefficient <= 1 && coefficient> 0 ) {
            this.coefficient = coefficient;
        } else {
            System.out.println("Not correct coefficient. Range: 0 > coef >= 1");
        }

    }

    @Override
    public double getSalary() {
        tmpSalary = super.getSalary()*coefficient;
        return tmpSalary;
    }
}
